### DISCUSSION TITLE
Hot reload calls OnDataContextChanged with the previous viewmodel (ShellViewModel)

### DISCUSSION THREAD

**Discussion by HakanL** [unoplatform/uno#12561] [Q&A] [answered]
My application is running fine, I load the shell (which displays a splash), then it navigates to my HomePage (which has a HomeViewModel) and everything is good. However, if I modify the XAML in VS which triggers a **hot reload** then I'm _losing all bound controls_. By adding a log entry to my OnDataContextChanged in HomePage, I can see that the data context gets replaced (from HomeViewModel) to **ShellViewModel** when hot reload is triggered. Is this a bug, or am I doing something wrong  (very possible)?

**Comment by jeromelaban** [comment, marked as answer]
HotReload on Uno.Extensions based apps is still in the works, and we've noticed those particular behaviors. @nickrandolph and the team are looking at enhancing that experience very soon.

**Reply by HakanL** [reply]
Ok, thanks for confirming!
